Test failures with python 3.5

```
======================================================================
ERROR: Check that pmap captures the errors when it is passed an object
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/lesteve/miniconda3/envs/py35/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/lesteve/dev/joblib/joblib/test/test_parallel.py", line 204, in test_parallel_pickling
    assert_raises(PickleError, Parallel(), (delayed(g)(x) for x in range(10)))
  File "/home/lesteve/miniconda3/envs/py35/lib/python3.5/unittest/case.py", line 724, in assertRaises
    return context.handle('assertRaises', args, kwargs)
  File "/home/lesteve/miniconda3/envs/py35/lib/python3.5/unittest/case.py", line 176, in handle
    callable_obj(*args, **kwargs)
  File "/home/lesteve/dev/joblib/joblib/parallel.py", line 805, in __call__
    while self.dispatch_one_batch(iterator):
  File "/home/lesteve/dev/joblib/joblib/parallel.py", line 658, in dispatch_one_batch
    tasks = BatchedCalls(itertools.islice(iterator, batch_size))
  File "/home/lesteve/dev/joblib/joblib/parallel.py", line 69, in __init__
    self.items = list(iterator_slice)
  File "/home/lesteve/dev/joblib/joblib/test/test_parallel.py", line 204, in <genexpr>
    assert_raises(PickleError, Parallel(), (delayed(g)(x) for x in range(10)))
  File "/home/lesteve/dev/joblib/joblib/parallel.py", line 162, in delayed
    pickle.dumps(function)
AttributeError: Can't pickle local object 'test_parallel_pickling.<locals>.g'

======================================================================
FAIL: Check that name collisions with functions will raise warnings
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/lesteve/miniconda3/envs/py35/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
AssertionError: 7 != 1
    '7 != 1' = '%s != %s' % _common_shorten_repr(7, 1)
    '7 != 1' = self._formatMessage('7 != 1', '7 != 1')
>>  raise self.failureException('7 != 1')
    

======================================================================
FAIL: Check that name collisions with functions will raise warnings
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/lesteve/miniconda3/envs/py35/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
AssertionError: False is not true
    'False is not true' = self._formatMessage('False is not true', "%s is not true" % safe_repr(False))
>>  raise self.failureException('False is not true')
    

======================================================================
FAIL: joblib.test.test_memory.test_memory_warning_lambda_collisions
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/lesteve/miniconda3/envs/py35/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/lesteve/dev/joblib/joblib/test/test_memory.py", line 254, in test_memory_warning_lambda_collisions
    nose.tools.assert_equal(len(w), 4)
AssertionError: 13 != 4
    '13 != 4' = '%s != %s' % _common_shorten_repr(13, 4)
    '13 != 4' = self._formatMessage('13 != 4', '13 != 4')
>>  raise self.failureException('13 != 4')
    
-------------------- >> begin captured logging << --------------------
root: WARNING: [Memory(cachedir='/tmp/lesteve/tmp3rc1u_3m/joblib')]: Flushing completely the cache
--------------------- >> end captured logging << ---------------------

======================================================================
FAIL: joblib.test.test_memory.test_memory_warning_collision_detection(11, 2)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/lesteve/miniconda3/envs/py35/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
AssertionError: 11 != 2
    '11 != 2' = '%s != %s' % _common_shorten_repr(11, 2)
    '11 != 2' = self._formatMessage('11 != 2', '11 != 2')
>>  raise self.failureException('11 != 2')
    
-------------------- >> begin captured logging << --------------------
root: WARNING: [Memory(cachedir='/tmp/lesteve/tmp3rc1u_3m/joblib')]: Flushing completely the cache
--------------------- >> end captured logging << ---------------------

======================================================================
FAIL: joblib.test.test_memory.test_memory_warning_collision_detection(False,)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/lesteve/miniconda3/envs/py35/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
AssertionError: False is not true
    'False is not true' = self._formatMessage('False is not true', "%s is not true" % safe_repr(False))
>>  raise self.failureException('False is not true')
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -7,6 +7,7 @@ env:
     - PYTHON_VERSION="3.3" NUMPY_VERSION="1.9"
     - PYTHON_VERSION="3.4" NUMPY_VERSION="1.9"
     - PYTHON_VERSION="3.4"
+    - PYTHON_VERSION="3.5" NUMPY_VERSION="1.9"
 
 install:
     - source continuous_integration/install.sh

--- a/joblib/test/test_memory.py
+++ b/joblib/test/test_memory.py
@@ -225,6 +225,7 @@ def test_memory_name_collision():
     with warnings.catch_warnings(record=True) as w:
         # Cause all warnings to always be triggered.
         warnings.simplefilter("always")
+        warnings.simplefilter("ignore", DeprecationWarning)
         a(1)
         b(1)
 
@@ -245,6 +246,7 @@ def test_memory_warning_lambda_collisions():
     with warnings.catch_warnings(record=True) as w:
         # Cause all warnings to always be triggered.
         warnings.simplefilter("always")
+        warnings.simplefilter("ignore", DeprecationWarning)
         nose.tools.assert_equal(0, a(0))
         nose.tools.assert_equal(2, b(1))
         nose.tools.assert_equal(1, a(1))
@@ -272,6 +274,7 @@ def test_memory_warning_collision_detection():
     with warnings.catch_warnings(record=True) as w:
         # Cause all warnings to always be triggered.
         warnings.simplefilter("always")
+        warnings.simplefilter("ignore", DeprecationWarning)
         a1(1)
         b1(1)
         a1(0)

--- a/joblib/test/test_memory.py
+++ b/joblib/test/test_memory.py
@@ -225,6 +225,9 @@ def test_memory_name_collision():
     with warnings.catch_warnings(record=True) as w:
         # Cause all warnings to always be triggered.
         warnings.simplefilter("always")
+        # This is a temporary workaround until we get rid of
+        # inspect.getargspec, see
+        # https://github.com/joblib/joblib/issues/247
         warnings.simplefilter("ignore", DeprecationWarning)
         a(1)
         b(1)
@@ -246,6 +249,9 @@ def test_memory_warning_lambda_collisions():
     with warnings.catch_warnings(record=True) as w:
         # Cause all warnings to always be triggered.
         warnings.simplefilter("always")
+        # This is a temporary workaround until we get rid of
+        # inspect.getargspec, see
+        # https://github.com/joblib/joblib/issues/247
         warnings.simplefilter("ignore", DeprecationWarning)
         nose.tools.assert_equal(0, a(0))
         nose.tools.assert_equal(2, b(1))
@@ -274,6 +280,9 @@ def test_memory_warning_collision_detection():
     with warnings.catch_warnings(record=True) as w:
         # Cause all warnings to always be triggered.
         warnings.simplefilter("always")
+        # This is a temporary workaround until we get rid of
+        # inspect.getargspec, see
+        # https://github.com/joblib/joblib/issues/247
         warnings.simplefilter("ignore", DeprecationWarning)
         a1(1)
         b1(1)

--- a/joblib/test/test_parallel.py
+++ b/joblib/test/test_parallel.py
@@ -201,7 +201,14 @@ def test_parallel_pickling():
     def g(x):
         return x ** 2
 
-    assert_raises(PickleError, Parallel(), (delayed(g)(x) for x in range(10)))
+    try:
+        # pickling a local function always fail
+        pickle.dumps(g)
+    except Exception as exc:
+        exception_class = exc.__class__
+
+    assert_raises(exception_class, Parallel(),
+                  (delayed(g)(x) for x in range(10)))
 
 
 def test_error_capture():

--- a/joblib/test/test_parallel.py
+++ b/joblib/test/test_parallel.py
@@ -202,7 +202,9 @@ def test_parallel_pickling():
         return x ** 2
 
     try:
-        # pickling a local function always fail
+        # pickling a local function always fail but the exception
+        # raised is a PickleError for python <= 3.4 and AttributeError
+        # for python >= 3.5
         pickle.dumps(g)
     except Exception as exc:
         exception_class = exc.__class__

